dev-lang/rust/rust-1.18.0.recipe

```diff
@@ -0,0 +1,217 @@
+CHECKSUM_SHA256_11="e1ad31337422f9b05f57f43266048935f246cc35ca53671feceb9e6208bc4b40"
+CHECKSUM_SHA256_12="927f6b782f0be32c6027335bbd49fb40cffa922095ffacc9a0aca78e45023f1a"
+SOURCE_DIR="rust-$srcGitRev"
+PATCHES="rust-$portVersion.patchset"
+PATCHES_3="rust-installer-$portVersion.patchset"
+PATCHES_7="cargo-$cargoVersion.patchset"
+PATCHES_11="rust-rls-$portVersion.patchset"
+PATCHES_12="cargo-vendor-$srcGitRev12.patchset"
+REVISION="1"
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	rust$secondaryArchSuffix = $portVersion
+	cmd:rustc = $portVersion
+	cmd:rustdoc = $portVersion
+	cmd:rust_gdb = $portVersion
+	cmd:rust_lldb = $portVersion
+	cmd:cargo$secondaryArchSuffix = $cargoVersion
+	cmd:rls$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcurl$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cargo$secondaryArchSuffix == 0.18.0
+	cmd:cmake
+	cmd:cmp
+	cmd:file
+	cmd:find
+	cmd:git
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python
+	cmd:rustc == 1.17.0
+	cmd:sed
+	cmd:tar
+	cmd:which
+	cmd:xargs
+	"
+
+BUILD()
+{
+	# With the new build system, Rust always makes a tarball with the source
+	# In older recipes we symlinked the other git submodules, now we have to
+	# copy them
+	rm -rf cargo src/compiler-rt src/llvm src/rust-installer \
+	       src/liblibc src/rt/hoedown src/doc/nomicon src/doc/reference \
+	       src/doc/book rls
+	cp -r /sources-2/compiler-rt-$srcGitRev2 src/compiler-rt
+	cp -r /sources-3/rust-installer-$srcGitRev3 src/rust-installer
+	cp -r /sources-4/hoedown-$srcGitRev4 src/rt/hoedown
+	cp -r /sources-5/libc-$srcGitRev5 src/liblibc
+	cp -r /sources-6/llvm-$srcGitRev6 src/llvm
+	cp -r /sources-7/cargo-$cargoVersion cargo
+	cp -r /sources-8/nomicon-$srcGitRev8 src/doc/nomicon
+	cp -r /sources-9/reference-$srcGitRev9 src/doc/reference
+	cp -r /sources-10/book-$srcGitRev10 src/doc/book
+	cp -r /sources-11/rls-$srcGitRev11 rls
+
+	# write the build configuration
+	cat >config.toml <<- EOL
+	[llvm]
+	targets = "X86"
+
+	[build]
+	cargo = "/$relativeBinDir/cargo"
+	rustc = "/boot/system/bin/rustc"
+	submodules = false
+	extended = true
+
+	[install]
+	prefix = "$prefix"
+	libdir = "$relativeLibDir"
+	mandir = "$relativeManDir"
+	docdir = "$relativeDevelopDocDir"
+
+	[rust]
+	channel = "stable"
+	use-jemalloc = false
+	rpath = false
+EOL
+
+	# the install step of rust depends on the cargo vendor command
+	# build it separately here
+	cd /sources-12/cargo-vendor-$srcGitRev12
+	cargo install --force
+
+	# now build rust and cargo
+	cd /sources/rust-$srcGitRev
```

Comment:
```diff
-	cd /sources/rust-$srcGitRev
+	cd "$sourceDir"
```

Reply:
Shouldn't it be $sourceDir/rust-$srcGitRev?

Reply:
haikuporter sets `sourceDir` to the absolute path to the `$SOURCE_DIR` directory (i.e. `rust-$srcGitRev`) so `cd "$sourceDir"` is fine.

(BTW, I tend to use quotes although they are not required for now (and will very likely never be) so I'm OK with not using quotes around `$sourceDir` (and friends), although I'd recommend using them in lines where there are already other changes being made.)